### PR TITLE
135-add-an-inbody-outbody bundle

### LINKED ISSUE #135: In-body out-body classification bundle
**Is your feature request related to a problem? Please describe.**
A in-body out-body classification bundle is a classfication bundle for endoscope scenario. This bundle should be able to discriminate whether a frame from endoscope video is in body. With the help of  this bundle, only in body videos can be used to do some extra works like detection and segmentation, in order to protect the privacy of patients.

**Describe the solution you'd like**
A in-body out-body classification bundle that can train, do inference and test on given datasets.

### PR DESCRIPTION
**Pull request by binliunls**
Signed-off-by: binliu <binliu@nvidia.com>

Fixes #135  .

### Description
Add an inbody-outbody bundle which can train, do inference and evaluation on given datasets.

### Status
**Work in progress**

### Please ensure all the checkboxes:

- [x] Codeformat tests passed locally by running `./runtests.sh --codeformat`.
- [ ] In-line docstrings updated.
- [ ] Update `version` and `changelog` in `metadata.json` if changing an existing bundle.
- [ ] Please ensure the naming rules in config files meet our requirements (please refer to: `CONTRIBUTING.md`).
- [ ] Ensure versions of packages such as `monai`, `pytorch` and `numpy` are correct in `metadata.json`.
- [ ] Descriptions should be consistent with the content, such as `eval_metrics` of the provided weights and TorchScript modules.
- [ ] Files larger than 25MB are excluded and replaced by providing download links in `large_file.yml`.
- [ ] Avoid using path that contains personal information within config files (such as use `/home/your_name/` for `"bundle_root"`).

### CODE REVIEW

File: models/inbody_outbody_classification/scripts/export_to_onnx.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    # compare ONNX Runtime and PyTorch results
+    np.testing.assert_allclose(numpy_torch_out, ort_outs[0], rtol=1e-03, atol=1e-05)
+    print("Exported model has been tested with ONNXRuntime, and the result looks good!")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    # the original model for converting.
+    parser.add_argument(
+        "--model", type=str, default=r"/workspace/bundle/DeID/models/model.pt", help="Input an existing model weight"
```
Comment (yiheng-wang-nv) [inline models/inbody_outbody_classification/scripts/export_to_onnx.py]: Hi @binliunls , here can the name of `DeID` be modified into this bundle's name?

Comment (Nic-Ma) [reply models/inbody_outbody_classification/scripts/export_to_onnx.py]: please don't use the word `DeID` in this bundle.
See the similar renaming in MONAI Label: https://github.com/Project-MONAI/MONAILabel/pull/1006.

Thanks.

Comment (binliunls) [reply models/inbody_outbody_classification/scripts/export_to_onnx.py]: Hi @Nic-Ma, @yiheng-wang-nv ,
I have update the path to workspace.

Thanks,
Bin

**Comment by Nic-Ma** [comment]
Hi @binliunls @yiheng-wang-nv @wyli ,

I would suggest the bundle name to "endoscopy_inbody_classification".
What do you think?

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Others look good to me except for the naming.

Thanks.

**Review by Nic-Ma — CHANGES_REQUESTED** [review]
Pendng for legal confirmation for the lisence.

Thanks.

**Comment by yiheng-wang-nv** [comment]
> Hi @binliunls @yiheng-wang-nv @wyli ,
> 
> I would suggest the bundle name to "endoscopy_inbody_classification". What do you think?
> 
> Thanks.

Hi @Nic-Ma , I'm fine with this name

File: models/endoscopy_inbody_classification/docs/readme.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    --maxShapes=INPUT__0:32x3x256x256 \
+    --shapes=INPUT__0:8x3x256x256
+```
+This command need TensorRT with correct CUDA installed in the environment. For the detail of installing TensorRT, please refer to [this link](https://docs.nvidia.com/deeplearning/tensorrt/install-guide/index.html).
+
+# References
+[1] J. Hu, L. Shen and G. Sun, Squeeze-and-Excitation Networks, 2018 IEEE/CVF Conference on Computer Vision and Pattern Recognition, 2018, pp. 7132-7141. https://arxiv.org/pdf/1709.01507.pdf
+
+# License
+This model is released under Apache 2.0 terms. The license file is included with the model.
```
Comment (yiheng-wang-nv) [inline models/endoscopy_inbody_classification/docs/readme.md]: Hi @binliunls , according to the discussion, the format of this part has been changed. Could you please modify it in two of your PRs? (you can refer to https://github.com/Project-MONAI/model-zoo/tree/dev/models/brats_mri_segmentation#license )

Comment (binliunls) [reply models/endoscopy_inbody_classification/docs/readme.md]: Hi @yiheng-wang-nv:
      I have updated license info in both endoscopy-inbody-classification and surgical-tool-segmentation bundles.
Thanks,
Bin

File: models/endoscopy_inbody_classification/docs/readme.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
+# Description
+A pre-trained model for the endoscopy inbody classification task.
+
+# Model Overview
+This model is trained using the SEResNet50 structure, whose details can be found in [1]. All datasets are from private samples of [Activ Surgical](https://www.activsurgical.com/). Samples in training and validation dataset are from the same 4 videos, while test samples are from different two videos.
+The [pytorch model](https://drive.google.com/file/d/14CS-s1uv2q6WedYQGeFbZeEWIkoyNa-x/view?usp=sharing) and [torchscript model](https://drive.google.com/file/d/1fOoJ4n5DWKHrt9QXTZ2sXwr9C-YvVGCM/view?usp=sharing) are shared in google drive.
```
Comment (jin-nvidia) [inline models/endoscopy_inbody_classification/docs/readme.md]: Suggest to add note:

Modify the path specified in `configs/train.json` and `configs/inference.json`: `"bundle_root"` to reflect where the models are downloaded. Under the path specified by `"bundle_root"` the expected directory structure is `models/model.pt`.

File: models/endoscopy_inbody_classification/docs/readme.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+Datasets used in this work were provided by [Activ Surgical](https://www.activsurgical.com/). Here is a [link](https://drive.google.com/uc?id=1rQQfHcZFs74OG0uJsL4vg7YqMWXV-s4k) of 20 samples (10 in-body and 10 out-body) to show what this dataset looks like.
+
+The input label json should be a list made up by dicts which includes "image" and "label" keys. An example format is shown below.
+
+```
+[
+    {
+        "image":"/path/to/image/image_name0.jpg",
+        "label": 0
+    }
```
Comment (jin-nvidia) [inline models/endoscopy_inbody_classification/docs/readme.md]: There needs to be a , after each }

File: models/endoscopy_inbody_classification/docs/readme.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
+# Description
+A pre-trained model for the endoscopy inbody classification task.
+
+# Model Overview
+This model is trained using the SEResNet50 structure, whose details can be found in [1]. All datasets are from private samples of [Activ Surgical](https://www.activsurgical.com/). Samples in training and validation dataset are from the same 4 videos, while test samples are from different two videos.
+The [pytorch model](https://drive.google.com/file/d/14CS-s1uv2q6WedYQGeFbZeEWIkoyNa-x/view?usp=sharing) and [torchscript model](https://drive.google.com/file/d/1fOoJ4n5DWKHrt9QXTZ2sXwr9C-YvVGCM/view?usp=sharing) are shared in google drive.
+
+## Data
+Datasets used in this work were provided by [Activ Surgical](https://www.activsurgical.com/). Here is a [link](https://drive.google.com/uc?id=1rQQfHcZFs74OG0uJsL4vg7YqMWXV-s4k) of 20 samples (10 in-body and 10 out-body) to show what this dataset looks like.
```
Comment (jin-nvidia) [inline models/endoscopy_inbody_classification/docs/readme.md]: Suggest to add note for user:
Modify the path specified in `configs/train.json` and `configs/inference.json`: `"dataset_dir"` to reflect where the data is.

File: models/endoscopy_inbody_classification/docs/readme.md (hunk 3 above)
Comment (binliunls) [reply models/endoscopy_inbody_classification/docs/readme.md]: Hi @jin-nvidia :
    Thanks for your advice! I have added a note of how to specify the model path in config files.
Thanks
BIn

File: models/endoscopy_inbody_classification/docs/readme.md (hunk 4 above)
Comment (binliunls) [reply models/endoscopy_inbody_classification/docs/readme.md]: Hi @jin-nvidia :
    Thanks for your advice! I have added commas in this part.
Thanks
BIn

File: models/endoscopy_inbody_classification/docs/readme.md (hunk 5 above)
Comment (binliunls) [reply models/endoscopy_inbody_classification/docs/readme.md]: Hi @jin-nvidia :
    Thanks for your advice! I have added a note of how to specify the data path in config files.
Thanks
BIn

File: models/endoscopy_inbody_classification/configs/inference.json [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    "preprocessing": {
+        "_target_": "Compose",
+        "transforms": [
+            {
+                "_target_": "LoadImaged",
+                "keys": "image"
+            },
+            {
+                "_target_": "ToTensord",
+                "keys": "label"
```
Comment (SachidanandAlle) [inline models/endoscopy_inbody_classification/configs/inference.json]: What label is doing here?

Comment (binliunls) [reply models/endoscopy_inbody_classification/configs/inference.json]: Hi @SachidanandAlle ,
      Thank you for your advice. There is no need of this process. I have deleted it.
Thanks,
BIn

**Review by Nic-Ma — APPROVED** [review]

**Review by yiheng-wang-nv — APPROVED** [review]
